igzip_042/include/types.h

```diff
@@ -0,0 +1,46 @@
+/**********************************************************************

```

Comment:
Possibly for 32-bit systems.  You'll find it hard to do 128bit integer maths on those as they don't have hardware support for a 128 bit register.